lib/SPIRV/libSPIRV/SPIRVValue.cpp

```diff
@@ -51,6 +51,14 @@ void SPIRVValue::setAlignment(SPIRVWord A) {
     eraseDecorate(DecorationAlignment);
     return;
   }
+  SPIRVWord PrevAlignment;
+  if (hasAlignment(&PrevAlignment)) {
+    // Do nothing if the Id already has an Alignment decoration, provided
+    // it matches the new alignment.
+    assert(A == PrevAlignment &&
```

Comment:
Mmm, some would argue, saying that if 2 sources provides different alignment, then we should use the biggest among them. I personally slightly leaning to not allow such case (aka keep assert here), but can imagine seeing complains about it. WDYT?

Reply:
That's the alternative approach I had considered, but it feels a bit wrong because different alignments kind of means there is a contradiction in the input.  So I first tried with an assert and the test suite passed.  If there is a good reason for allowing different alignments then I'm happy to drop the assert.

Reply:
Well, good thing about such assertion - we can always immediately get a notification of such use case and remove the assertion if necessary, so lets keep it as is :)